src/components/Tutorial/TutorialScrollUpButton.tsx

```diff
@@ -0,0 +1,45 @@
+import { FC, useCallback } from 'react'
+import scrollTo from '../../device/scrollTo'
+import scrollTopStore from '../../stores/scrollTop'
+import TutorialNavigationButton from './TutorialNavigationButton'
+
+/** A button to scroll up to the tutorial. */
+const TutorialScrollUpButton: FC<{ show: boolean }> = ({ show }) => {
+  const scrollTop = scrollTopStore.useState()
```

Comment:
For performance reasons, let's ensure the component is only re-rendered on scroll when visible.

You might try something like this:

```js
const scrollTop = scrollTopStore.useSelector(state => show ? 0 : state)
```

However my brief testing shows that the component is still getting re-rendered on scroll, so it may be coming from a parent also. Maybe the component needs `React.memo`?

Reply:
Rendering when scrolling and not-visible is necessary for the animation of the component. If we don't position it correctly before/after it being shown, then it does not transition to/from the correct location.

Changing the value of `scrollTop` to be `null` with that selector does not prevent the `scrollTopStore` from still triggering the re-render, and `React.memo` can only help with `props`, not the renders triggered within the component by the store.

Fortunately, I don't believe this should cause any worthwhile performance issues; if there were issues when the element is not displayed, then there would also be issues when the element is displayed. With the `scrollTopStore` already handling throttling and this element only being rendered during the brief tutorial, pursuing this particular case may be premature optimization.

Reply:
That's a fair point. I'm fine leaving it as-is then.